### PR TITLE
Update and polish for README and DocC landing page

### PR DESCRIPTION
**Pull request by simonjbeaumont**
### Motivation

Ahead of 1.0, we want to give the README and DocC landing page a refresh.

### Modifications

Updates to the README and DocC landing page.

### Result

Updates to the README and DocC landing page.

### Test Plan

Review.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,60 +8,171 @@ Generate Swift client and server code from an OpenAPI document.
...
-## Repository organization
+The code is generated at build-time, so it's always in sync with the OpenAPI document and doesn't need to be committed to your source repository.
 
-The Swift OpenAPI Generator project is split across multiple repositories to enable extensibility and minimize dependencies in your project.
+## Features
+
+- Works with OpenAPI Specification versions 3.0 and 3.1.
+- Streaming request and response bodies enabling use cases such as JSON event streams, and large payloads without buffering.
+- Support for JSON, multipart, URL-encoded form, base64, plain text, and raw bytes, represented as value types with type-safe properties.
+- Client, server and middleware abstractions, decoupling the generated code from the HTTP client library and web framework.
```
Comment (czechboy0) [inline README.md:22]:
```suggestion
- Client, server, and middleware abstractions, decoupling the generated code from the HTTP client library and web framework.
```

File: README.md [hunk 2]
```diff
@@ -8,60 +8,171 @@ Generate Swift client and server code from an OpenAPI document.
...
+
+The generated `Client` type provides a method for each HTTP operation defined in the OpenAPI document[^example-openapi-yaml] and can be used with any HTTP library that provides an implementation of `ClientTransport`.
+
+```swift
+import OpenAPIURLSession
+import Foundation
+
+let client = Client(
+  serverURL: URL(string: "http://localhost:8080/api")!,
+  transport: URLSessionTransport()
```
Comment (czechboy0) [inline README.md:42]:
```suggestion
    serverURL: URL(string: "http://localhost:8080/api")!,
    transport: URLSessionTransport()
```

Just to make the spaces consistent with the server example below.

File: Sources/swift-openapi-generator/Documentation.docc/Swift-OpenAPI-Generator.md [hunk 3]
```diff
@@ -8,49 +8,90 @@ Generate Swift client and server code from an OpenAPI document.
...
 
-**swift-openapi-generator** ([source][repo-generator], [docs][docs-generator]) provides the plugin.
+```swift
+import OpenAPIURLSession
+import Foundation
 
-**swift-openapi-runtime** ([source][repo-runtime], [docs][docs-runtime]) provides a library with common types and abstractions used by the generated code.
+let client = Client(
+  serverURL: URL(string: "http://localhost:8080/api")!,
+  transport: URLSessionTransport()
```
Comment (czechboy0) [inline Sources/swift-openapi-generator/Documentation.docc/Swift-OpenAPI-Generator.md:42]:
```suggestion
    serverURL: URL(string: "http://localhost:8080/api")!,
    transport: URLSessionTransport()
```

Same as above.

File: Sources/swift-openapi-generator/Documentation.docc/Swift-OpenAPI-Generator.md [hunk 4]
```diff
@@ -8,49 +8,90 @@ Generate Swift client and server code from an OpenAPI document.
...
 The code is generated at build-time, so it's always in sync with the OpenAPI document and doesn't need to be committed to your source repository.
 
-### Getting started
+## Features
 
-> Tip: In a rush? Get started quickly by <doc:Checking-out-an-example-project>.
+- Works with OpenAPI Specification versions 3.0 and 3.1.
+- Streaming request and response bodies enabling use cases such as JSON event streams, and large payloads without buffering.
+- Support for JSON, multipart, URL-encoded form, base64, plain text, and raw bytes, represented as value types with type-safe properties.
+- Client, server and middleware abstractions, decoupling the generated code from the HTTP client library and web framework.
```
Comment (czechboy0) [inline Sources/swift-openapi-generator/Documentation.docc/Swift-OpenAPI-Generator.md:22]:
```suggestion
- Client, server, and middleware abstractions, decoupling the generated code from the HTTP client library and web framework.
```

**Review by czechboy0 — APPROVED** [review]
Just a few tiny nits, otherwise looks great!

File: README.md (hunk 1 above)
Comment (simonjbeaumont) [reply README.md]: Done.

File: README.md (hunk 2 above)
Comment (simonjbeaumont) [reply README.md]: Done.

File: Sources/swift-openapi-generator/Documentation.docc/Swift-OpenAPI-Generator.md (hunk 4 above)
Comment (simonjbeaumont) [reply Sources/swift-openapi-generator/Documentation.docc/Swift-OpenAPI-Generator.md]: Done.

File: Sources/swift-openapi-generator/Documentation.docc/Swift-OpenAPI-Generator.md (hunk 3 above)
Comment (simonjbeaumont) [reply Sources/swift-openapi-generator/Documentation.docc/Swift-OpenAPI-Generator.md]: Done.

**Pull request merged**